Set max session idle time (#642)
fauna/faunadb-js: README.md

This Driver supports and is tested on:

- Node.js [_Current_, _Active LTS_, and _Maintenance LTS_ releases](https://nodejs.org/en/about/releases/)
  - _Current_ - v17
  - _Active LTS_ - currently v16

>   - _Maintenance LTS_ - v12 and v14

Reviewer: why the formatting changes?

Author: VS code auto formatter started doing stuff. Should be fixed now

When running on the Node.js platform, the Fauna client uses [HTTP/2 multiplexing](https://stackoverflow.com/questions/36517829/what-does-multiplexing-mean-in-http-2)
to reuse the same session for many simultaneous requests. After all open requests
have been resolved, the client will keep the session open for a period of time

> to be reused for any new requests.

Reviewer: please leave the 500 ms in - that is still valid.

The `http2SessionIdleTime` parameter may be used to control how long the HTTP/2

> session remains open while the connection is idle. To save on the overhead of

Reviewer 2: ```suggestion
session remains open while the query connection is idle. To save on the overhead of
```

closing and re-opening the session, set `http2SessionIdleTime` to a longer time.

> The default value is 500ms and the maximum value is 5000ms.

Reviewer 2: ```suggestion
The default value is 500ms and the maximum value is 5000ms.

Note that `http2SessionIdleTime` has no effect on a stream connection: a stream
is a long-lived connection that is intended to be held open indefinitely.
```

> While an HTTP/2 session is alive, the client will hold the Node.js event loop

Reviewer 2: ```suggestion
While an HTTP/2 session is alive, the client holds the Node.js event loop
```

async function main() {
  const client = new Client({
    secret: 'YOUR_FAUNADB_SECRET',
    http2SessionIdleTime: 1000,

>     //                    ^^^ Non-negative integer

Reviewer 2: ```suggestion
    http2SessionIdleTime: 1000, // Must be a non-negative integer
```

Reviewer 2: Note that the comment on line 309 is no longer relevant, since `Infinity` is no longer being used.
